src/System.IO.Pipelines/System/IO/Pipelines/Pipe.cs

```diff
@@ -363,7 +344,7 @@ internal ValueAwaiter<FlushResult> FlushAsync(CancellationToken cancellationToke
         /// <param name="exception">Optional Exception indicating a failure that's causing the pipeline to complete.</param>
         void IPipeWriter.Complete(Exception exception)
         {
-            if (_writingState.IsActive)
+            if (_writingState.IsStarted && _currentWriteLength > 0)
```

Comment:
We should just remove the IsStarted/IsActive flag and use the current write length for everything

Reply:
We still need it for some things - Advance before calling GetMemory, and to prevent returning of the last segment if it was returned from GetMemory and might be written to.

Reply:
Hmm, I wonder if other IOutput implementations also have this problem